lib/variable/slice.cpp

```diff
@@ -48,12 +48,40 @@ auto get_coord(const Variable &coord, const Dim dim) {
   return std::tuple(coord, ascending);
 }
 
+void expect_same_unit(const Variable &coord, const Variable &value,
+                      const std::string &name) {
+  if (coord.unit() != value.unit()) {
+    throw except::UnitError("The unit of the " + name + " of the slice (" +
+                            to_string(value.unit()) +
+                            ") does not match the unit of the coordinate (" +
+                            to_string(coord.unit()) + ").");
+  }
+}
+
+void expect_valid_dtype(const Variable &var, const std::string &name) {
+  if (!is_total_orderable(var.dtype())) {
+    throw except::TypeError(
+        "The dtype of the " + name + " of the slice (" +
+        to_string(var.dtype()) +
+        ") cannot be used for value-based slicing because it does not"
+        " define an order.");
```

Comment:
Strictly speaking this is just an detail of they current implementation. Nothing fundamentally prevents supporting this for non-orderable types.

Reply:
How would this select a range if the coord cannot be ordered?

Reply:
Selecting a range would be impossible, but getting a single item would (think of, e.g., a string key).